internal/extensionapi/connection_route.go

```diff
@@ -14,9 +61,140 @@ func (s *ExtensionServer) HandleConnectionCreate(w http.ResponseWriter, r *http.
 		return
 	}
 
-	logger.Info("Handling Connection creation", "method", r.Method, "path", r.URL.Path)
+	// Extract namespace from URL path
+	pathParts := strings.Split(r.URL.Path, "/")
+	if len(pathParts) < 6 {
+		WriteError(w, http.StatusBadRequest, "Invalid URL path")
+		return
+	}
+	namespace := pathParts[5] // /apis/connection.workspaces.jupyter.org/v1alpha1/namespaces/{namespace}/connections
+
+	// Parse request body
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		WriteError(w, http.StatusBadRequest, "Failed to read request body")
+		return
+	}
+
+	var req WorkspaceConnectionRequest
+	if err := json.Unmarshal(body, &req); err != nil {
+		WriteError(w, http.StatusBadRequest, "Invalid JSON")
+		return
+	}
+
+	// Validate request
+	if err := validateWorkspaceConnectionRequest(&req); err != nil {
+		WriteError(w, http.StatusBadRequest, err.Error())
+		return
+	}
+
+	// Check if CLUSTER_ARN is configured early for VSCode connections
+	if req.Spec.WorkspaceConnectionType == ConnectionTypeVSCodeRemote {
+		eksClusterARN := os.Getenv(aws.EKSClusterARNEnv)
```

Comment:
pass from [config](https://github.com/jupyter-infra/jupyter-k8s/blob/fe7fc5dd43f9602ad16758c091407515046f9332/internal/extensionapi/config.go#L19) for consistency please (in an `// aws section`)